doc/specs/stdlib_stringlist.md

```diff
@@ -0,0 +1,487 @@
+---
+title: stringlist
+---
+# Lists of strings
```

Comment:
What is the difference with the [`stdlib` `stringlist_type`](https://github.com/fortran-lang/stdlib/blob/master/doc/specs/stdlib_stringlist_type.md)?